_posts/2021-05-26-registry-security.md

```diff
@@ -0,0 +1,64 @@
+---
+layout: post
+title: "Securing Apicurio Registry 2.0 using Keycloak"
+date:   2021-05-28 12:00:00
+author: carles
+categories: blog registry security
+---
+
+
+Hey everyone, in Apicurio Registry 2.0.0.Final we've introduced support for some security concepts. In this blog post, I will try to explain them and show you how to configure them.
+
+Core Features
+===
+Essentially in this blog post, we will cover the following:
+
+* **Authentication based on Keycloak** - optionally protect the registry such that the API requires users to authenticate (BASIC and OAuth supported)
+* **Role based authorization** - when authentication is enabled, you can also enable this feature, and users will need to have at least one role of (sr-admin, sr-developer, sr-readonly)
+* **Creator-only authorization option** - option to prevent changes to artifacts unless the authenticated user is the user who originally created the artifact
+
+
+Keycloak Configuration
+===
+
+### Authentication based on Keycloak
+In order to secure Apicurio Registry you will need a Keycloak server up and running along with some specific configuration.
+Essentially, you will need to create a Realm, two clients, and some roles (if you want to use roles).  By default Apicurio Registry expects the name of the Realm to be `registry`.
+When security is enabled, Apicurio Registry will expect two Keycloak clients.  By default, these clients should be named `registry-api` and `apicurio-registry` (but don't worry, you will be able to customize everything by using environment variables!)
+Below you have the required configuration for the clients.
+
+![Configure Apicurio Registry API_Client](/images/guides/registry-auth-client-api-config.png)
+
+![Configure Apicurio Registry_UI Client](/images/guides/registry-auth-client-ui-config.png)
+
+You will need to pay attention to the configuration - especially to the access types and the configuration URLs. 
+For the `registry-api` client, make sure to configure it as bearer-only. For the `apicurio-registry` client, you will need to configure it as public, to allow users to connect to your UI.
+You will also need to configure Web Origins, you can use `+` to allow Web Origins coming from the redirect URL.    
+
+Now, you will need to tweak the Apicurio Registry configuration to get this to work.
+
+![Configure Apicurio Registry Auth_Properties](/images/guides/registry-auth-properties-config.png)
+
+The first property that you will need to use is `AUTH_ENABLED`.  You will need to set it to true to enable the integration.  The second property you need to set is the Keycloak server url `KEYCLOAK_URL`.
+We provide some sensible defaults for most of the configuration, but please, note that you can customize most of them by using environment variables.
+
+### Role based authorization
+
+To enable roles, you need to set the `ROLES_ENABLED` propert to `true`. 
```

Comment:
Suggest "Role-based authorization"

Suggest "To enable roles, set the `ROLES_ENABLED` property" (no need to say "you need" :) and missing "y" in propert")